common/types.go

```diff
@@ -16,6 +16,14 @@ func NewBigInt(v int64) BigInt {
 	return BigInt{*big.NewInt(v)}
 }
 
+func (b BigInt) MarshalText() ([]byte, error) {
+	return []byte(b.String()), nil
+}
+
+func (b *BigInt) UnmarshalText(text []byte) error {
+	return b.Int.UnmarshalJSON(text)
```

Comment:
Should this be

```suggestion
	return b.Int.UnmarshalText(text)
```

Reply:
yes! I intentionally made it JSON, I remember cargo-culting it from somewhere and thinking it was weird ... IDK what happened. Anyway, tested with UnmarshalText now, works as expected. Thanks.